schemas/definitions.json

```diff
@@ -229,8 +229,7 @@
 
     "url": {
       "type": "string",
-      "description": "A valid URL.",
-      "pattern": "^(http(s?):.*|mailto:.*|tel:(\\+?)[0-9]{4,})$"
+      "description": "A location of a resource, potentially relative. For hierarchical schemes (like http), applications must resolve relative URIs (e.g. './v1/api/'). Implementations should support the following schemes: http:, https:, mailto:, tel:, and ws:."
```

Comment:
@tkurki The regex pattern for URLs only describes a small subset of potential ones and does not reflect the actual syntax allowed. For example, the original `(http(s?):.*` fragment only says http(s) is a valid scheme but does not enforce any of the restrictions of what's allowed in the scheme-specific part. With the escaping and I10N aspects of valid URL syntax I did not think the regex added much.

Relative URLs would avoid having to generate full absolute values for references to resources on the same server. For example, the discovery result could be
```json
{
  "endpoints": {
    "v1": {
      "vers, ion": "1.1.2",
      "signalk-http": "/signalk/v1/api/",
    }
  }
}
```
which has the advantage of being constant for any installation of a server it it's following our convention. Actually, all same the advantages as for using relative references in HTML.

Resolution would be done [as defined by the RFC](https://tools.ietf.org/html/rfc3986#section-5.2), so if that data was returned by a GET to `https://10.0.1.1/signalk` then the resolved path would be `https://10.0.1.1/signalk/v1/api/`